### PR TITLE
Removes dependencies between tests

### PR DESCRIPTION
**Pull request by allonsy**
Introduces a database with sample users for deterministic, non interdependent testing.
I also included a quick fix to kill the server using node's child process library.
This addresses #4 and #8 
@mjhm @wgottschalk let me know what you think

### CODE REVIEW

File: example/testapp/apitest/features/support/feature_hooks.js [hunk 1]
```diff
@@ -7,15 +7,15 @@ module.exports = function () {
   var self = this;
 
   this.registerHandler('BeforeFeatures', function (event, done) {
-    childProcess.exec("../src/serv")
+    this.serverPID = childProcess.exec("../src/serv")
```
Comment (wgottschalk-RMS) [inline example/testapp/apitest/features/support/feature_hooks.js]: the `this` keyword is bound to the global object here. maybe `self.serverPID` so the process is attached to the module?

Comment (allonsy) [inline example/testapp/apitest/features/support/feature_hooks.js]: Ah, that make the declaration on line 7 make more sense.

**Comment by allonsy** [comment]
changed `this` to `self`

File: example/testapp/apitest/features/step_definitions/steps.js [hunk 2]
```diff
@@ -3,7 +3,11 @@ var exec = require('sync-exec')
 var cucumberApi = require('../../../../../index.js');
 
 resetDatabase = function() {
-  exec("../src/serv --reset-db")
+  exec("cp ../src/db/empty.db ../src/db/test.db")
```
Comment (mjhm) [inline example/testapp/apitest/features/step_definitions/steps.js:6]: Ideally this shouldn't be a step but rather should be in the `support/feature_hooks.js` within the `BeforeFeatures` hook.  Then the `truncateTables` function should be in the main `Before` hook within `support/scenario_hooks.js`.

**Comment by mjhm** [comment]
All good. Merge if you like or implement the db setup in the hooks.  We'll likely be revisiting the database setup multiple times.

**Pull request closed without merging**